R-package/tests/testthat/test_parameters.R

```diff
@@ -101,14 +101,83 @@ test_that("training should warn if you use 'dart' boosting, specified with 'boos
         , num_threads = .LGB_MAX_THREADS
     )
     params[[boosting_param]] <- "dart"
+
+    # expect warning
+    # if early_stopping_rounds is specified
     expect_warning({
       result <- lightgbm(
         data = train$data
         , label = train$label
         , params = params
         , nrounds = 5L
         , verbose = -1L
+        , early_stopping_rounds = 1
       )
     }, regexp = "Early stopping is not available in 'dart' mode")
+
+    # expect no warning
+    # if early_stopping_rounds is not
+    # specified see:
+    # https://github.com/microsoft/LightGBM/issues/6612
+    expect_no_warning({
+      result <- lightgbm(
+        data = train$data
+        , label = train$label
+        , params = params
+        , nrounds = 5L
+        , verbose = -1L,
```

Comment:
```suggestion
        , verbose = -1L
```

There's an extra comma here. I suspect this means you didn't run these tests locally before submitting this. Please do that.

```shell
sh build-cran-package.sh --no-build-vignettes
R CMD INSTALL ./lightgbm*.tar.gz
cd R-package/tests
Rscript testthat.R
```

Reply:
Hi @jameslamb,

I didn't run the tests locally, this is true. For what it's worth, I tried building the package with the changes I have made with the code you provided - but it will not build it with the additions that I have made. So I was relying on the `workflows` to do the checking.

And on an honest sidenote, I was under the impression that draft `PR`s were only visible to me.

Reply:
> I was under the impression that draft PRs were only visible to me.

Ah totally understandable! But no, they're visible to everyone on the project.

If you'd like me to stop reviewing until you've taken this out of draft and marked it ready for review, please let me know. This is not urgent, so I'd be happy to back off and give you some space to test things. That's how we usually work in this repo (draft PRs are not reviewed), but I have seen some contributors who open a draft PR and expect a review, so I assumed that was what you wanted here.

Reply:
Allow me to mark it ready for review! :smile: Man, I am learning so much from this simple task. Thank you for your patience!